test/test_va_api_config_attribs.cpp

```diff
@@ -39,40 +40,106 @@ class VAAPIConfigAttribs
     {
         doTerminate();
     }
+
+    void validateConfigAttributes(const ConfigAttributes& actual,
+        const ConfigAttributes& supported) const
+    {
+        const size_t size(actual.size());
+        ASSERT_EQ(size, supported.size());
+
+        // Require that actual and supported are in the same order, by type
+        for (size_t i(0); i < size; ++i) {
+            const VAConfigAttrib& aAttrib = actual[i];
+            const VAConfigAttrib& sAttrib = supported[i];
+
+            ASSERT_EQ(aAttrib.type, sAttrib.type);
+
+            switch (aAttrib.type) {
+            case VAConfigAttribRTFormat:
+                EXPECT_EQ(sAttrib.value & aAttrib.value, aAttrib.value);
+                break;
+
+            // The following type's return a bitfield of supported choices
```

Comment:
Just to make a note: not all aAttrib.type are covered under this switch case. There are other bit field cases which needs to be covered. ie. VAConfigAttribFEIFunctionType and more. This should be in #ToDo list.

Reply:
There is a question as to whether these test requirements are correct to begin with.  That is, is it legal for driver to return the same bitfield value (i.e. all supported bitmasks) in `vaQueryConfigAttributes` as returned from `vaGetConfigAttributes` if user/app did not explicitly specify a attribute value during `vaCreateConfig`.  Currently, iHD driver does this and fails these tests (for the currently covered bitfield attribute types) since this test expects `vaQueryConfigAttributes` to return the *actual* attribute value being used.

For example, `vaGetConfigAttributes` might return `VAConfigAttribRateControl=VA_RC_VBR | VA_RC_CBR | VA_RC_CQP` and user might call `vaCreateConfig` without `VAConfigAttribRateControl` attribute set.  In this case, driver should choose *one* default value for `VAConfigAttribRateControl`.  Question is, when `vaQueryConfigAttributes` is called for that config, should the driver return the default `VAConfigAttribRateControl` value it chose during `vaCreateConfig` or should it return all possible supported values as returned from `vaGetConfigAttributes` (or just not return a `VAConfigAttribRateControl` at all when not specified by user/app)?

Reply:
> ...Question is, when vaQueryConfigAttributes is called for that config, should the driver return the default VAConfigAttribRateControl value it chose during vaCreateConfig or should it return all possible supported values as returned from vaGetConfigAttributes (or just not return a VAConfigAttribRateControl at all when not specified by user/app)?

After a little more investigation, both i965 and iHD do the latter two (of 3) behaviors.  i965 just chooses the last behavior for most attributes, which is why it passes more tests than iHD.  Thus, I'll change this test to behave like the current "default" case for all attributes, if that seems reasonable.